File: src/store/projects.ts
import { create } from 'zustand';
import type { Project } from '../lib/types';
import * as api from '../lib/tauri';

interface ProjectsState {
  projects: Project[];
  currentProject: Project | null;
  isLoading: boolean;
  error: string | null;

  loadProjects: () => Promise<void>;
  selectProject: (project: Project | null) => void;
  createProject: (name: string) => Promise<Project>;
  deleteProject: (projectId: string) => Promise<void>;
}

export const useProjectsStore = create<ProjectsState>((set) => ({
  projects: [],
  currentProject: null,
  isLoading: false,
  error: null,

  loadProjects: async () => {
    set({ isLoading: true, error: null });
    try {
      const projects = await api.listProjects();
      set({ projects, isLoading: false });
    } catch (error) {
      set({ error: String(error), isLoading: false });
    }
  },

  selectProject: (project) => {
    set({ currentProject: project });
  },

  createProject: async (name) => {
    set({ isLoading: true, error: null });
    try {
      const project = await api.createProject(name);
      set((state) => ({
        projects: [project, ...state.projects],
        isLoading: false,
      }));
      return project;
    } catch (error) {
      set({ error: String(error), isLoading: false });
      throw error;
    }
  },

  deleteProject: async (projectId) => {
    set({ isLoading: true, error: null });
    try {
      await api.deleteProject(projectId);
      set((state) => ({
        projects: state.projects.filter((p) => p.id !== projectId),
        currentProject: state.currentProject?.id === projectId ? null : state.currentProject,
        isLoading: false,
      }));
    } catch (error) {
      set({ error: String(error), isLoading: false });
      throw error;
    }
  },
}));

File: src-tauri/src/commands/projects.rs
use crate::db;
use crate::db::models::{Project, ProjectSettings};
use crate::error::Result;
use crate::state::AppState;
use tauri::State;

#[tauri::command]
pub async fn list_projects(state: State<'_, AppState>) -> Result<Vec<Project>> {
    let projects = db::list_projects(&state.db).await?;
    Ok(projects)
}

#[tauri::command]
pub async fn create_project(state: State<'_, AppState>, name: String) -> Result<Project> {
    let project = db::create_project(&state.db, name).await?;
    Ok(project)
}

#[tauri::command]
pub async fn get_project(state: State<'_, AppState>, project_id: String) -> Result<Project> {
    let project = db::get_project(&state.db, &project_id).await?;
    Ok(project)
}

#[tauri::command]
pub async fn get_project_settings(
    state: State<'_, AppState>,
    project_id: String,
) -> Result<ProjectSettings> {
    let project = db::get_project(&state.db, &project_id).await?;
    let settings: ProjectSettings = serde_json::from_str(&project.settings_json)?;
    Ok(settings)
}

#[tauri::command]
pub async fn update_project_settings(
    state: State<'_, AppState>,
    project_id: String,
    settings: ProjectSettings,
) -> Result<()> {
    db::update_project_settings(&state.db, &project_id, settings).await?;
    Ok(())
}

#[tauri::command]
pub async fn delete_project(state: State<'_, AppState>, project_id: String) -> Result<()> {
    db::delete_project(&state.db, &project_id).await?;
    Ok(())
}
